tools/ci_set_matrix.py

```diff
@@ -228,23 +229,34 @@ def get_settings(board):
         set_output(f"boards-{arch}", json.dumps(sorted(arch_to_boards[arch])))
 
 
-def set_docs_to_build(build_all):
-    if "build-doc" in last_failed_jobs:
-        build_all = True
-
-    doc_match = build_all
-    if not build_all:
-        doc_pattern = re.compile(
-            r"^(?:.github/workflows/|docs|extmod/ulab|(?:(?:ports/\w+/bindings|shared-bindings)\S+\.c|conf\.py|tools/extract_pyi\.py|requirements-doc\.txt)$)|(?:-stubs|\.(?:md|MD|rst|RST))$"
-        )
-        for p in changed_files:
-            if doc_pattern.search(p):
-                doc_match = True
-                break
+def set_docs_to_build(build_doc: bool):
+    if not build_doc:
+        if "build-doc" in last_failed_jobs:
+            build_doc = True
+        else:
+            doc_pattern = re.compile(
+                r"^(?:.github/workflows/|docs|extmod/ulab|(?:(?:ports/\w+/bindings|shared-bindings)\S+\.c|conf\.py|tools/extract_pyi\.py|requirements-doc\.txt)$)|(?:-stubs|\.(?:md|MD|rst|RST))$"
```

Comment:
Escape the `.` in `.github`:
```suggestion
                r"^(?:\.github/workflows/|docs|extmod/ulab|(?:(?:ports/\w+/bindings|shared-bindings)\S+\.c|conf\.py|tools/extract_pyi\.py|requirements-doc\.txt)$)|(?:-stubs|\.(?:md|MD|rst|RST))$"
```